src/asciidoc/docs/develop/testingTemplates.adoc

```diff
@@ -0,0 +1,164 @@
+For instance, you can manually enter an HTTP request, configure a header for it, send the request, and examine the results returned by the server.
+
+=== Previewing different contexts
+
+A Koji can display different landing pages depending on the context in which it is running.
+For more on contexts, see <<withkoji-vcc-package#_feedsdk#>>.
+
+When running a Koji outside the editor, you can display its landing page by appending a query string to the URL.
+For instance, if your Koji has an admin page, you can display it by appending `?context=admin` to the URL.
+
+Inside the editor, the *Live preview* tab will display buttons for the following contexts if they exist in your Koji: *Admin*, *About*, *Sticker*, *Buyer Receipt*, *Seller Receipt*.
+Clicking the button for one of these contexts displays a preview of its landing page.
+
+You can use other names for custom contexts, but *Live preview* will not display buttons for them.
+You will have to preview them by using the query string method.
+
+=== Writing to the console
+
+You can test JavaScript code by using the `console.log` method, which prints to the console.
+To see the console in the Koji editor, open the *Thumbnail* tab and click on the console button:
+image:consoleButton.jpg[Console button]
+
+For example, the following code snippet prints *Key pressed* to the console every time you press a key, as long as the *Live preview* pane is selected.
+
+[NOTE]
+You can select *Live preview* by clicking anywhere in its pane.
+
+[source,javascript]
+----
+    $(window).keydown(function (e) {
+        KEY_STATUS.keyDown = true;
+        if (KEY_CODES[e.keyCode]) {
+            e.preventDefault();
+            KEY_STATUS[KEY_CODES[e.keyCode]] = true;
+        }
+        console.log("Key pressed");
+    }).keyup(function (e) {
+        KEY_STATUS.keyDown = false;
+        if (KEY_CODES[e.keyCode]) {
+            e.preventDefault();
+            KEY_STATUS[KEY_CODES[e.keyCode]] = false;
+        }
+    });
+----
+
+When running your Koji outside the editor, either in production or as a preview, you can access the console through the browser's menu.
+(See <<testing-templates#_using_browser_tools_for_testing,Using browser tools for testing>>.)
+
+=== Testing In-app purchases
+
+You can test *In-app purchases* by using *Debug in live player* on the *Remote* tab.
+
+. First, make sure you've topped up your *Wallet* in *KojiPay*.
+You'll be using the Koji to send money to yourself, so you can issue yourself a refund when you finish testing.
+
+. Run your Koji in *Debug in live player* and try paying yourself a small amount.
+If all goes well, you'll see this amount deducted from your *Wallet*.
+The transaction will be marked *pending*.
+
+. You should be able to refund the transaction, resetting your balance to its original value and cancelling the transaction fee.
+
+[NOTE]
+If *KojiPay* doesn't allow you to refund the transaction, wait a minute or two and try again.
+It may not be finished processing.
```

Comment:
This note is very helpful. The last sentence has odd phrasing. Maybe you can work the concept into the previous sentence?

Reply:
I wasn't sure how to work it into a single sentence, but see what you think of the new phrasing.

Reply:
That works, thanks!